client/ayon_core/hosts/houdini/plugins/create/create_hda.py

```diff
@@ -68,21 +90,103 @@ def create_instance_node(
         else:
             hda_node = to_hda
 
-        hda_node.setName(node_name)
+        # If user tries to create the same HDA instance more than
+        # once, then all of them will have the same product name and
+        # point to the same hda_file_name. But, their node names will
+        # be incremented.
+        hda_node.setName(node_name, unique_name=True)
         self.customize_node_look(hda_node)
+
+        # Set Custom settings.
+        hda_def = hda_node.type().definition()
+
+        if pre_create_data.get("use_ayon_icon"):
+            hda_def.setIcon(get_ayon_icon_filepath())
+
+        if pre_create_data.get("set_user"):
+            hda_def.setUserInfo(getpass.getuser())
```

Comment:
Replace `import getpass` with `from ayon_core.lib import get_ayon_username`
```suggestion
            hda_def.setUserInfo(get_ayon_username())
```